### DISCUSSION TITLE
How to change the animation speed of an indeterminate ProgressBar?

### DISCUSSION THREAD

**Discussion by alessandrofrancesconi** [AvaloniaUI/Avalonia#14047] [Q&A]
I'm using an indeterminate ProgressBar:

`<ProgressBar IsIndeterminate="False" />`

How can I control the animation speed of the bar?

**Comment by Pratham95VP** [comment]
I don't think there is built in property for directly controlling the animation speed. You can experiment with animating the ProgressBar by adjusting its properties and incorporating time control. While there's no guarantee of success, even if it does work, you may need to fine-tune the implementation as the desired effect might not be as smooth as anticipated. If you came across any findings please update in the same thread

**Comment by thevortexcloud** [comment]
You can probably override the animation by setting your own style. This is how it's done in the Simple theme

https://github.com/AvaloniaUI/Avalonia/blob/fe671b2cebf6f0a4b55e1a3a5a55cd7969c1122a/src/Avalonia.Themes.Simple/Controls/ProgressBar.xaml#L84-L113

If that does not work, you can override the actual control template. But you need to copy in a functioning template to modify (EG one from one of the themes) and it may suddenly stop working properly if Avalonia updates.
